megamek/src/megamek/client/ui/swing/PlanetaryConditionsDialog.java

```diff
@@ -148,6 +149,24 @@ public void windowClosing(WindowEvent e) {
             }
         });
 
+        choWeather.addActionListener(e -> {
+            int weather = choWeather.getSelectedIndex();
+            if (currentWeather != weather && 
+                    (weather == PlanetaryConditions.WE_LIGHT_HAIL ||
+                     weather == PlanetaryConditions.WE_HEAVY_HAIL ||
+                     weather == PlanetaryConditions.WE_LIGHT_SNOW || 
+                     weather == PlanetaryConditions.WE_SLEET ||
+                     weather == PlanetaryConditions.WE_SNOW_FLURRIES ||
+                     weather == PlanetaryConditions.WE_HEAVY_SNOW ||
+                     weather == PlanetaryConditions.WE_ICE_STORM || 
+                     weather == PlanetaryConditions.WE_BLIZZARD ||
+                     weather == PlanetaryConditions.WE_MOD_SNOW)) {
+                currentWeather = weather;
```

Comment:
`currentWeather` only gets updated if set to winter conditions. If you change from snow to non-snow, `currentWeather` remains set to the snow conditions. So if I go to light snow, then adjust the temp from -40 to 5, then clear, I get different behavior if I set it back to light snow vs. a different snow condition. While seems likely to confuse and frustrate users. in addition to the weird snow temperature rules that we didn't make up to begin with.

Reply:
So, moving `currentWeather = weather;` outside of the `if` statement produces an unwanted side effect. I think it produces the dynamic behavior we're looking for, but the temperature doesn't stick. I think it's getting set correctly when the dialog box is closed, it's just that the field get's reset when you open the dialog back up, so it looks like it isn't working (need to test that assumption further). That being said, I think the question to ask is, what is the desired behavior? I think refactoring the dialog would be easier if we have a clear cut idea of how it's supposed to function based on the funky weather rules.

Reply:
In that case I think this is probably enough at this point. Looking at this makes me want to rewrite all the planetary conditions using enums but I'm not going to do that right now.